### ISSUE TITLE
False negative for blanket `noqa` (PGH004)

### ISSUE THREAD

**Issue by mthuurne** [astral-sh/ruff#5371]
If I run Ruff on this code:
```py
foo  # noqab
```
with these arguments and no config file:
```
ruff check --select F,PGH004 testcase.py
```
it reports no errors.

I would expect Ruff to report either `F821`, because `foo` is undefined, or `PGH004`, because no rule code was specified.

This seems to happen whenever there is a non-whitespace character after `noqa`. If there is nothing or whitespace after `noqa`,  `PGH004` is reported as expected.

Tested on Ruff 0.0.274.

Some context: I originally had `# noqa[SLF001]` in my code, in an attempt to specify a rule, but accidentally using the mypy suppression syntax instead of the flake8 suppression syntax. Fortunately I realized this mistake soon, but otherwise the blanket suppression could have hidden unexpected errors, which is exactly what `PGH004` seeks to prevent.

This could be fixed by implementing more of #850, but if that is going to take a while, perhaps there could be a dedicated fix for this issue, by either ignoring the `noqa` in cases like this or reporting `PGH004`.

**Comment by charliermarsh** [comment]
Unfortunately, this is consistent with Flake8. If you use `# ruff: noqa`, it'll act as you expect: we _won't_ ignore the code, and we'll show you a warning:

```console
❯ ruff check foo.py
warning: Unexpected suffix on `noqa` directive: "# ruff: noqa[]"
foo.py:1:8: F401 [*] `os` imported but unused
```

**Comment by charliermarsh** [comment]
(Preemptively closing because, while confusing, I think this is expected for Flake8 compatibility, and the recommended solution is to use the Ruff-specific suppression comment, but happy to keep discussing of course.)
